--- a/libavformat/Makefile	2017-06-07 20:40:05.995269967 +0200
+++ b/libavformat/Makefile	2017-06-07 20:42:50.035131075 +0200
@@ -566,12 +566,12 @@
 OBJS-$(CONFIG_MMST_PROTOCOL)             += mmst.o mms.o asf.o
 OBJS-$(CONFIG_PIPE_PROTOCOL)             += file.o
 OBJS-$(CONFIG_PROMPEG_PROTOCOL)          += prompeg.o
-OBJS-$(CONFIG_RTMP_PROTOCOL)             += rtmpproto.o rtmppkt.o
-OBJS-$(CONFIG_RTMPE_PROTOCOL)            += rtmpproto.o rtmppkt.o
-OBJS-$(CONFIG_RTMPS_PROTOCOL)            += rtmpproto.o rtmppkt.o
-OBJS-$(CONFIG_RTMPT_PROTOCOL)            += rtmpproto.o rtmppkt.o
-OBJS-$(CONFIG_RTMPTE_PROTOCOL)           += rtmpproto.o rtmppkt.o
-OBJS-$(CONFIG_RTMPTS_PROTOCOL)           += rtmpproto.o rtmppkt.o
+OBJS-$(CONFIG_FFRTMP_PROTOCOL)             += rtmpproto.o rtmppkt.o
+OBJS-$(CONFIG_FFRTMPE_PROTOCOL)            += rtmpproto.o rtmppkt.o
+OBJS-$(CONFIG_FFRTMPS_PROTOCOL)            += rtmpproto.o rtmppkt.o
+OBJS-$(CONFIG_FFRTMPT_PROTOCOL)            += rtmpproto.o rtmppkt.o
+OBJS-$(CONFIG_FFRTMPTE_PROTOCOL)           += rtmpproto.o rtmppkt.o
+OBJS-$(CONFIG_FFRTMPTS_PROTOCOL)           += rtmpproto.o rtmppkt.o
 OBJS-$(CONFIG_RTP_PROTOCOL)              += rtpproto.o
 OBJS-$(CONFIG_SCTP_PROTOCOL)             += sctp.o
 OBJS-$(CONFIG_SRTP_PROTOCOL)             += srtpproto.o srtp.o
--- a/libavformat/protocols.c	2017-06-07 20:40:05.996269942 +0200
+++ b/libavformat/protocols.c	2017-06-07 20:45:58.584373798 +0200
@@ -44,12 +44,12 @@
 extern const URLProtocol ff_md5_protocol;
 extern const URLProtocol ff_pipe_protocol;
 extern const URLProtocol ff_prompeg_protocol;
-extern const URLProtocol ff_rtmp_protocol;
-extern const URLProtocol ff_rtmpe_protocol;
-extern const URLProtocol ff_rtmps_protocol;
-extern const URLProtocol ff_rtmpt_protocol;
-extern const URLProtocol ff_rtmpte_protocol;
-extern const URLProtocol ff_rtmpts_protocol;
+extern const URLProtocol ff_ffrtmp_protocol;
+extern const URLProtocol ff_ffrtmpe_protocol;
+extern const URLProtocol ff_ffrtmps_protocol;
+extern const URLProtocol ff_ffrtmpt_protocol;
+extern const URLProtocol ff_ffrtmpte_protocol;
+extern const URLProtocol ff_ffrtmpts_protocol;
 extern const URLProtocol ff_rtp_protocol;
 extern const URLProtocol ff_sctp_protocol;
 extern const URLProtocol ff_srtp_protocol;
--- a/libavformat/rtmpproto.c    2016-10-01 03:12:42.000000000 +0200
+++ b/libavformat/rtmpproto.c    2016-10-13 10:55:16.862635525 +0200
@@ -2578,7 +2578,7 @@
 static int rtmp_open(URLContext *s, const char *uri, int flags)
 {
     RTMPContext *rt = s->priv_data;
-    char proto[8], hostname[256], path[1024], auth[100], *fname;
+    char *proto, tmpProto[10], hostname[256], path[1024], auth[100], *fname;
     char *old_app, *qmark, *n, fname_buffer[1024];
     uint8_t buf[2048];
     int port;
@@ -2590,7 +2590,7 @@
 
     rt->is_input = !(flags & AVIO_FLAG_WRITE);
 
-    av_url_split(proto, sizeof(proto), auth, sizeof(auth),
+    memset(tmpProto, 0, sizeof(tmpProto)); proto = &tmpProto[2]; av_url_split(tmpProto, sizeof(tmpProto), auth, sizeof(auth),
                  hostname, sizeof(hostname), &port,
                  path, sizeof(path), s->filename);
 
@@ -3124,9 +3124,9 @@
 };
 
 
-RTMP_PROTOCOL(rtmp)
-RTMP_PROTOCOL(rtmpe)
-RTMP_PROTOCOL(rtmps)
-RTMP_PROTOCOL(rtmpt)
-RTMP_PROTOCOL(rtmpte)
-RTMP_PROTOCOL(rtmpts)
+RTMP_PROTOCOL(ffrtmp)
+RTMP_PROTOCOL(ffrtmpe)
+RTMP_PROTOCOL(ffrtmps)
+RTMP_PROTOCOL(ffrtmpt)
+RTMP_PROTOCOL(ffrtmpte)
+RTMP_PROTOCOL(ffrtmpts)
